### PR TITLE
[WIP] fix: S3 URLs (#1373)

### PR DESCRIPTION
**Pull request by Sukram21**
Generate correct S3 urls that won't result in SSL certificate warnings anymore. Outlined in #1373.

### CODE REVIEW

File: packages/electron-builder-http/src/publishOptions.ts [hunk 1]
```diff
@@ -115,7 +120,8 @@ export interface S3Options extends PublishConfiguration {
 }
 
 export function s3Url(options: S3Options) {
-  let url = `https://${options.bucket}.s3.amazonaws.com`
+  let url = `https://s3.${options.region}.amazonaws.com/${options.bucket}`
+
   if (options.path != null) {
```
Comment (develar) [inline packages/electron-builder-http/src/publishOptions.ts]: As this URL format can leads to issues, let's use it only bucket name contains dots.

File: packages/electron-builder/src/publish/PublishManager.ts [hunk 2]
```diff
@@ -89,6 +90,12 @@ export class PublishManager implements PublishContext {
         }
       }
 
+      if (publishConfig.provider === "s3") {
+        const s3 = new S3({signatureVersion: "v4"})
```
Comment (develar) [inline packages/electron-builder/src/publish/PublishManager.ts]: As commented above, we should do it only if bucket name contains dot — avoid net calls.

**Comment by develar** [comment]
@Sukram21 please see other my comments.

I have fixed the issue related to API doc generation, diff should be small and reasonable.

**Comment by Sukram21** [comment]
Pushed a new commit dealing with your comments 🙂

**Comment by develar** [comment]
Will be published tomorrow. I don't want to add aws-sdk dep to main module, but it requires some publisher API changes. It is the reason why s3 publisher is not part of electron-builder module.

**Comment by develar** [comment]
Thanks a lot for PR, it makes s3 publisher more user-friendly.

**Comment by Sukram21** [comment]
You're welcome, I'm glad to contribute to this awesome project :)

**Pull request merged**